pallets/score-staking/src/lib.rs

```diff
@@ -445,6 +478,53 @@ pub mod pallet {
 			let payment = Scores::<T>::get(&account).ok_or(Error::<T>::UserNotExist)?;
 			Self::claim(origin, payment.unpaid_reward)
 		}
+
+		#[pallet::call_index(9)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn update_token_staking_amount(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			amount: BalanceOf<T>,
+			round_index: RoundIndex,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin.clone())?;
+			ensure!(
+				T::TokenStakingAuthorizer::can_update_staking(&sender),
+				Error::<T>::UnauthorizedOrigin
+			);
+
+			match Scores::<T>::get(&account) {
+				Some(mut s) => {
+					if round_index <= s.last_token_distributed_round {
+						return Err(Error::<T>::TokenStakingAmountAlreadyUpdated.into());
+					}
+					let last_round = Round::<T>::get();
+					s.token_staking_amount = amount;
+					s.last_token_distributed_round = last_round.index;
+					Scores::<T>::insert(account.clone(), s);
+					Self::deposit_event(Event::TokenStakingAmountUpdated {
+						account: account.clone(),
+						amount,
+					});
+					Ok(Pays::No.into())
+				},
+				None => Err(Error::<T>::UserNotExist.into()),
+			}
+		}
+
+		#[pallet::call_index(10)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn complete_reward_distribution(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
```

Comment:
Hmm strange - where is the token distribution process? (I thought it would be done within this extrinsic)

Reply:
This is just to trigger an event that marks the distribution for the round as completed. (Currently we don't do anything with the `token_staking_amount`, will there be another phase/issue to come?)

Reply:
No I mean "distribution for the round as completed" should be triggered by the worker - when it finishes with staking amount upload for all accounts, it should trigger the distribution

Reply:
I have updated the code to make the distribution on `update_token_staking_amount`, this extrinsic will be kept to signal that the distribution for the round is completed.